### PR TITLE
Workflow improvements

### PR DESCRIPTION
**Pull request by FP22FD**
## Summary
I improved the workflow adding some tools, integration tests with Cypress and unit tests with jest. 

## Checklist
- [x] configured the project with: eslint, prettier, commit hooks, GitHub Actions & secrets
- [x] configure the project for Jest and Cypress
- [x] added unit tests and integration tests for login, logout and ui/auth file
- [x] added the readme.me 
- [x] record a bug in the "issue" tab [#428](https://github.com/NoroffFEU/social-media-client/issues/428) with a fix solution in worfklow-fixes branch.

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
 name: Automated E2E Testing
-on: 
+on:
   pull_request:
   workflow_dispatch:
 
+env:
+  ## Sets environment variable
+
+  APP_HOME_URL: ${{secrets.APP_HOME_URL}}
```
Comment (Fermain) [inline .github/workflows/e2e-test.yml:9]: Beautiful

File: cypress/jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (Fermain) [inline cypress/jsconfig.json:1]: I'm not familiar with this file

File: cypress/support/commands.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+/// <reference types="Cypress"/>
+
+//Ref. https://github.com/cypress-io/cypress/issues/249
+if (Cypress.env("DELAY_COMMANDS")) {
```
Comment (Fermain) [inline cypress/support/commands.js:32]: I haven't seen this before either

File: jsconfig.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (Fermain) [inline jsconfig.json:1]: I'm not convinced you need these files

**Review by Fermain — APPROVED** [review]
Lovely submission, one of the best in class

**Comment by FP22FD** [comment]
Sorry @Fermain for the delay, I was busy with the SP2 delivery and forgot to check this. Thank you for your kind words. I tried my best.

**Pull request closed without merging**